rules/aws_security_group_rule_invalid_protocol.go

```diff
@@ -0,0 +1,95 @@
+package rules
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/terraform-linters/tflint-plugin-sdk/hclext"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+	"github.com/terraform-linters/tflint-ruleset-aws/project"
+)
+
+// AwsSecurityGroupRuleInvalidProtocolRule checks whether "protocol" has invalid value
+type AwsSecurityGroupRuleInvalidProtocolRule struct {
+	tflint.DefaultRule
+
+	resourceType  string
+	attributeName string
+	protocols     map[string]struct{}
+}
+
+// NewAwsSecurityGroupRuleInvalidProtocolRule returns new rule with default attributes
+func NewAwsSecurityGroupRuleInvalidProtocolRule() *AwsSecurityGroupRuleInvalidProtocolRule {
+	return &AwsSecurityGroupRuleInvalidProtocolRule{
+		resourceType:  "aws_security_group_rule",
+		attributeName: "protocol",
+		protocols: map[string]struct{}{
+			"tcp":    {},
+			"udp":    {},
+			"icmp":   {},
+			"icmpv6": {},
```

Comment:
`all` is also accepted as protocol.
https://github.com/hashicorp/terraform-provider-aws/blob/724c62ba7cea20d512187d83db1ed384080a3e92/internal/service/ec2/vpc_security_group.go#L1330-L1344

Reply:
Thank you for reviewing my PR!
I fixed it. e050a3c95f18fad648d68bb68b192c8ea0290102